### PR TITLE
New pipeline to release api-stub-generator to devops feed

### PR DESCRIPTION
**Pull request by praveenkuttappan**
New pipeline to build, test and release python apiview parser

### CODE REVIEW

**Review by tjprescott — COMMENTED** [review]
This looks great to me but since I'm not a devops guy I'll leave it to one of you all to approve :)

File: eng/pipelines/templates/stages/apiview-python-ci.yml
Comment (weshaggard) [inline eng/pipelines/templates/stages/apiview-python-ci.yml:28]: Why is this going here instead of the azure-sdk-for-python DevOps feed?

File: eng/pipelines/templates/stages/apiview-python-ci.yml
Comment (weshaggard) [reply eng/pipelines/templates/stages/apiview-python-ci.yml:100]: Should we be using our use-python-version template?

File: eng/pipelines/templates/stages/apiview-python-ci.yml
Comment (weshaggard) [inline eng/pipelines/templates/stages/apiview-python-ci.yml:41]: When did we add this template to the tools repo? Is this something worth sharing with the python repo?

File: eng/pipelines/templates/stages/apiview-python-ci.yml
Comment (weshaggard) [inline eng/pipelines/templates/stages/apiview-python-ci.yml:11]: I would still love for us to get these tools somewhere under the tools directory.

File: eng/pipelines/templates/stages/apiview-python-ci.yml
Comment (scbedd) [reply eng/pipelines/templates/stages/apiview-python-ci.yml:100]: Default if you don't specify is `3.X`.  That's about as much as I can predictably tell you.

Comment (scbedd) [reply eng/pipelines/templates/stages/apiview-python-ci.yml:100]: No real need, but probably should yeah.

 `use-python-version.yml` is used when you're uncertain the version of python you're going after is actually present on the build machine.

This looks like a slightly older version. I will file an issue to move the full version (along with pypy hacks) that python currently uses into `eng/common`. 

It's definitely useful on more than just the python repo.

Comment (scbedd) [reply eng/pipelines/templates/stages/apiview-python-ci.yml:100]: https://github.com/Azure/azure-sdk-tools/issues/3064

File: eng/pipelines/templates/stages/apiview-python-ci.yml
Comment (scbedd) [reply eng/pipelines/templates/stages/apiview-python-ci.yml:41]: https://github.com/Azure/azure-sdk-tools/issues/3064

File: eng/pipelines/templates/stages/apiview-python-ci.yml
Comment (scbedd) [reply eng/pipelines/templates/stages/apiview-python-ci.yml]: With the latest commit I'm making this a template that we can call from a different package.

**Review by scbedd — APPROVED** [review]
I have adjusted this to be non-package-specific. [Here is an example build](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1483511&view=logs&j=826b5e63-3908-5d05-b7a1-4a06c78abfa8)

File: packages/python-packages/api-stub-generator/ci.yml
Comment (weshaggard) [inline packages/python-packages/api-stub-generator/ci.yml:29]: Still curious if this is the best feed? We are trying to consume fewer feeds so I would suggest using the azure-sdk-for-python feed unless there is a reason not to.

File: packages/python-packages/api-stub-generator/ci.yml
Comment (weshaggard) [inline packages/python-packages/api-stub-generator/ci.yml:42]: Is the package name api-stub-generator as well? I'm mostly asking because we generally want to use a correctly namespaced name as we start to publish packages. I don't know what the actual conventionis for python but for .NET we use Microsoft.* and JS we use @azure-tools etc but something that indicates it is from our team or MS.

File: eng/pipelines/templates/stages/archetype-sdk-tool-python.yml
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-sdk-tool-python.yml:22]:
```suggestion
  - template: /eng/pipelines/templates/variables/globals.yml
```

File: packages/python-packages/api-stub-generator/ci.yml
Comment (praveenkuttappan) [reply packages/python-packages/api-stub-generator/ci.yml:29]: I was thinking of grouping a separate feed for apiview tools. But I am fine to update this to azure-sdk-for-python feedas well. @scbedd Let me know what you think.

File: packages/python-packages/api-stub-generator/ci.yml
Comment (praveenkuttappan) [reply packages/python-packages/api-stub-generator/ci.yml:42]: It is api-stub-generator. We can make a separate change to update this name to azure-apiview-stub-generator

**Review by praveenkuttappan — COMMENTED** [review]
Looks good.

File: packages/python-packages/api-stub-generator/ci.yml
Comment (scbedd) [reply packages/python-packages/api-stub-generator/ci.yml:29]: I don't mind! I'll update that before merging.

Comment (scbedd) [reply packages/python-packages/api-stub-generator/ci.yml]: [Build Targeting Python Feed](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1483829&view=results)

File: packages/python-packages/api-stub-generator/ci.yml
Comment (weshaggard) [inline packages/python-packages/api-stub-generator/ci.yml:35]: We should use PythonVersion here.

File: packages/python-packages/api-stub-generator/ci.yml
Comment (weshaggard) [reply packages/python-packages/api-stub-generator/ci.yml:42]: Yes we should come up with some naming that is less generic.

**Pull request merged**
